components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/dao/WorkflowRequestDAO.java

```diff
@@ -126,6 +126,68 @@ public WorkflowRequest retrieveWorkflow(String uuid) throws InternalWorkflowExce
         return null;
     }
 
+    /**
+     * Retrieve the External Workflow Identifier from Workflow ID.
+     *
+     * @param workflowId workflow ID retrieve from workflow request
+     * @return External Workflow Identifier
+     * @throws InternalWorkflowException
+     */
+    public String retrieveExternalWorkflowId(String workflowId) throws InternalWorkflowException {
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement prepStmt = null;
+        ResultSet rs = null;
+
+        String query = SQLConstants.GET_WORKFLOW_ID_QUERY;
+        try {
+            prepStmt = connection.prepareStatement(query);
+            prepStmt.setString(1, workflowId);
+            rs = prepStmt.executeQuery();
+            if (rs.next()) {
+                return rs.getString(SQLConstants.PARAM_VALUE_COLUMN);
+            }
+        } catch (SQLException e) {
+            throw new InternalWorkflowException("Error when executing the sql query:" + query, e);
```

Comment:
improve this :  handleException(WFConstant.Exceptions.SQL_ERROR_GETTING_WORKFLOW_COUNT, e);